src/esbmc/bmc.cpp

```diff
@@ -667,21 +564,19 @@ smt_convt::resultt bmct::run_thread(std::shared_ptr<symex_target_equationt> &eq)
 
   catch(std::string &error_str)
   {
-    message_streamt message_stream(*get_message_handler());
-    message_stream.error(error_str);
+    msg.error(error_str);
     return smt_convt::P_ERROR;
   }
 
   catch(const char *error_str)
   {
-    message_streamt message_stream(*get_message_handler());
-    message_stream.error(error_str);
+    msg.error(error_str);
     return smt_convt::P_ERROR;
   }
 
   catch(std::bad_alloc &)
   {
-    std::cout << "Out of memory" << std::endl;
+    msg.error("Out of memory\n");
```

Comment:
When does this `Out of memory` happen? If we know, we could say in which stage of the verification process it happens (e.g., symbolic execution, SMT solving, etc).

Reply:
That's the problem of having these catch-all exception handling, there is no way of knowing where they came from.

Reply:
> When does this Out of memory happen? If we know, we could say in which stage of the verification process it happens (e.g., symbolic execution, SMT solving, etc).

If you want to know specifically where it happened you need to create inner try-catches. For example, during Z3 solving if it catches the `bad_alloc` then you can throw an Z3_bad_alloc (which inherites std::bad_alloc) exception and then try again with a different strategy/solver. Thats the idea of exception handling.

> That's the problem of having these catch-all exception handling, there is no way of knowing where they came from.

The best approach would be to specialize everything that you want to handle differently. However, if you intend to debug the issue it won't make any difference. Most debuggers will stop if an exception is thrown (specially, bad_alloc). Also, if the exception is not handled, the default behaviour is to throw an abort anyway.